### PR TITLE
chansrv improved config support

### PR DESCRIPTION
**Pull request by matt335672**
This pull request:-
- Creates a config module for chansrv similar to the one in sesman.
- adds an optional -c (or --config) option to chansrv to allow the path to `sesman.ini` to be specified.
- removes the `CHANSRV_RESTRICT_OUTBOUND_CLIPBOARD` environment variable used between sesman and chansrv to pass `RestrictOutboundClipboard` over - it's now read directly from the config file.

This is one of three PRs to implement #1588. This one does not depend on the other two and has two other uses:-
1. It could be easily expanded to read chansrv logging parameters from sesman.ini (See #1633) 
2. Removing `CHANSRV_RESTRICT_OUTBOUND_CLIPBOARD`  allows for a manual chansrv restart process to be implemented as follows:-
   - unmount ~/thinclient_drives if mounted.
   - Run chansrv directly from the command line in the same session
   - Disconnect and reconnect so xrdp picks up the new chansrv.

   At present this is only likely to be of use to developers as it allows for chansrv to be run under a debugger or valgrind, etc.

I've also tidied up the manpage and made some minor changes to the cleanup code in chansrv.c

### CODE REVIEW

**Comment by metalefty** [comment]
Overall, I agree with this but I'm not sure giving `sesman.ini` for `chansrv` configuration is the best way. 
Functionally, it has no problem but a little bit misleading.

File: docs/man/xrdp-chansrv.8.in [hunk 1]
```diff
@@ -3,7 +3,7 @@
 \fBxrdp\-chansrv\fR \- \fBxrdp\fR channel server
 
 .SH "SYNTAX"
-.B xrdp\-chansrv
+.B xrdp\-chansrv [  --config /path/to/sesman.ini ]
```
Comment (metalefty) [inline docs/man/xrdp-chansrv.8.in:6]: If not specified, what will happen? Refer `XRDP_CFG_PATH/sesman.ini`?

File: sesman/chansrv/chansrv.c [hunk 2]
```diff
@@ -1792,32 +1765,54 @@ main(int argc, char **argv)
     enum logReturns error;
     struct log_config logconfig;
     enum logLevels log_level;
-    char *restrict_outbound_clipboard_env;
     g_init("xrdp-chansrv"); /* os_calls */
 
+    g_snprintf(default_sesman_ini, 255, "%s/sesman.ini", XRDP_CFG_PATH);
```
Comment (metalefty) [inline sesman/chansrv/chansrv.c:1770]: I found this.

File: docs/man/xrdp-chansrv.8.in (hunk 1 above)
Comment (metalefty) [reply docs/man/xrdp-chansrv.8.in:6]: It looks so.

Comment (matt335672) [reply docs/man/xrdp-chansrv.8.in:6]: I meant for the square brackets [] to indicate it's an optional value.

Regarding your previous comment there are other possibilities:-
1. we can use environment variables to pass the chansrv configuration from sesman.
2. We can pass the sesman.ini path to chansrv via environment variable
3. We could set up a separate chansrv config file.

I'm not keen on the first option. It complicates adding variables and makes a chansrv restart complicated. I'm happy with either of the other two if you'd prefer. Or is there another option?

File: sesman/chansrv/chansrv.c (hunk 2 above)
Comment (matt335672) [reply sesman/chansrv/chansrv.c:1770]: That's the default value if nothing is specified. I'll add a comment and I can also remove the hard-coded limit.

**Comment by metalefty** [comment]
> - removes the `CHANSRV_RESTRICT_OUTBOUND_CLIPBOARD` environment variable used between sesman and chansrv to pass `RestrictOutboundClipboard` over - it's now read directly from the config file.

This sounds pretty good but there is concern.

System administrator wants to `RestrictOutboundClipboard` not to bringing out any data from remote xrdp session to local client.  Some corporate users have such requirements.  Usually, `/etc/xrdp/sesman.ini` is  editable only by root user. As a result, users cannot turn off `RestrictOutboundClipboard` by themselves.

I think users can turn off `RestrictOutboundClipboard` switch, right? 

```
cp /etc/xrdp/sesman.ini $HOME/sesman.ini
sed -i .bak -e 's/RestrictOutboundClipboard=true/RestrictOutboundClipboard=false/' sesman.ini
pkill xrdp-chansrv
xrdp-chansrv --config $HOME/sesman.ini
```

In short, `RestrictOutboundClipboard` must not be overridden by users other than system administrator.

**Comment by metalefty** [comment]
I've just noticed environment variable is modifiable by user.  So passing via `CHANSRV_RESTRICT_OUTBOUND_CLIPBOARD` is a bad idea. It should be read directly from the config file. It is my design mistake.

**Comment by matt335672** [comment]
They're both good points. Leave it with me and I'll see what I can do.

**Comment by matt335672** [comment]
Having had a think it seems the simplest option is simply to remove the ability for the user to easily change the sesman.ini file in use.

There's not a lot of config used by chansrv, and the things the user might want to override (logging levels) can be done by environment variable anyway.

So now this PR simply tidies up the chansrv config code, and removes the dependency on the `CHANSRV_RESTRICT_OUTBOUND_CLIPBOARD` variable.

File: docs/man/xrdp-chansrv.8.in [hunk 3]
```diff
@@ -28,16 +28,32 @@ Remote Applications Integrated Locally
 Dynamic Virtual Channel
 .RS
 
+.SH ENVIRONMENT
+.TP
+.I CHANSRV_LOG_LEVEL
+Logging level for chansrv from 0 (minimal) to 5 (extensive). Defaults to 3.
```
Comment (metalefty) [inline docs/man/xrdp-chansrv.8.in:34]: I usually specify the log level by name. Never tried numeric values, it will work though.

**Comment by metalefty** [comment]
Overall, LGTM.

**Comment by matt335672** [comment]
Thanks. I'll look at squashing and merging.

After your comment about the logging level I've had a look at the code. It seems the standard logging code in `common/log.c` does indeed accept numbers, but chansrv doesn't at the moment. For now I'll just update the manpage to match what the code does. I don't want to make a lot of changes here until we've merged #1633. 

I'll also make some changes to #1650 to make it clear that if the user wants to use a different config for xrdp-sesman, they will still need to provide a basic sesman.ini for xrdp and xrdp-chansrv.

**Review by metalefty — APPROVED** [review]
Approved. Merge at your convenience.
